yargs.js

```diff
@@ -669,6 +692,42 @@ function Yargs (processArgs, cwd, parentRequire) {
     return detectLocale
   }
 
+  var hasOutput = false
+  var exitError = null
+  // maybe exit, always capture
+  // context about why we wanted to exit.
+  self.exit = function (code, err) {
+    hasOutput = true
+    exitError = err
+    if (exitProcess) process.exit(code)
+  }
+
+  // we use a custom logger that buffers output,
+  // so that we can print to non-CLIs, e.g., chat-bots.
+  var silent = false
+  var _logger = {
+    log: function () {
+      var args = Array.prototype.slice.call(arguments)
+      if (!silent) console.log.apply(console, args)
+      hasOutput = true
+      output += args.join(' ')
+    },
+    error: function () {
+      var args = Array.prototype.slice.call(arguments)
+      if (!silent) console.error.apply(console, args)
+      hasOutput = true
+      output += args.join(' ')
+    }
+  }
+  self._getLoggerInstance = function () {
+    return _logger
+  }
+  // has yargs output an error our help
+  // message in the current execution context.
+  self._hasOutput = function () {
+    return !!hasOutput
```

Comment:
Since this is a private variable that is always a boolean, we could get away with just `return hasOutput` here.